feat: Introduce page & colleciton level SEO and make the product and … (#539)
Shopify/hydrogen: packages/hydrogen/src/framework/docs/seo.md

## How SEO works

> The Hydrogen framework work with Seo by inspects the user-agent for every request, and buffer the response to fully render the response on server side instead.

Reviewer: ```suggestion
The Hydrogen framework supports SEO by inspecting the `user-agent` for every request, and buffering the response to fully render it on server-side instead.
```

To emulate this behaviour, add `?_bot` query param at the end of url.

> ## What Hydrogen provides

Reviewer: ```suggestion
## How SEO works in Hydrogen
```

> The Hydrogen framework supplies a `<DefaultSeo>` server component, which is responsible for fetching your `shop.name` and `shop.description` to act as default seo values for every page on your website.

Reviewer: ```suggestion
The Hydrogen framework supplies a `<DefaultSeo>` server component, which fetches your `shop.name` and `shop.description`, and provides the default SEO values for every page on your website.
```

> The Hydrogen package also includes a `<Seo>` client component that are currently included in the starter template on home page, product page, collection page, and pages page server entry components that output the SEO-related tags in your document `head` and override the default values.

Reviewer: Maybe we can simplify this and also link out to the various examples in `/hydrogen`? For example, we can link to  `packages/hydrogen/src/components/Seo/CollectionSeo.client.tsx` for the collection page.

```suggestion
Hydrogen also includes a `<Seo>` client component in the starter template that you can use to output the SEO-related tags in your document `head` and override the default values on the following pages: 

- Home page
- Collection page
- Product page
...
```

## How SEO works

The Hydrogen framework supports SEO by inspecting the `user-agent` for every request, and buffering the response to fully render it on server-side instead.

> To emulate this behaviour, add `?_bot` query param at the end of url.

Author: Mostly for testing if developer want to see what the bot would see since the SEO robot flow would be different from normal.

> To emulate this behaviour, add `?_bot` query param at the end of url.

Reviewer: What do you think of adding something like: 

"To test whether a bot can crawl the content of a webpage, add the `?_bot` query parameter at the end of the webpage's URL."

## How SEO works in Hydrogen

The Hydrogen framework supplies a `<DefaultSeo>` server component, which fetches your `shop.name` and `shop.description`, to provides the default SEO values for every page on your website.

Hydrogen also includes a `<Seo>` client component in the starter template that you can use to output the SEO-related tags in your document `head` and override the default values on the following pages:

- [Default page](../../../../../examples/template-hydrogen-default/src/components/DefaultSeo.server.jsx)
- [Home page](../../../../../examples/template-hydrogen-default/src/pages/index.server.jsx)
- [Product page](../../../../../examples/template-hydrogen-default/src/pages/products/[handle].server.jsx)
- [Collection page](../../../../../examples/template-hydrogen-default/src/pages/collections/[handle].server.jsx)

> - [Pages page](../../../../../examples/template-hydrogen-default/src/pages/pages/[handle].server.jsx)

Author: Sure thing!
